django/db/models/fields/related.py

```diff
@@ -681,7 +681,7 @@ def contribute_to_related_class(self, cls, related):
         # Internal FK's - i.e., those with a related name ending with '+' -
         # and swapped models don't get a related descriptor.
         if not self.remote_field.is_hidden() and not related.related_model._meta.swapped:
-            setattr(cls, related.get_accessor_name(), self.related_accessor_class(related))
+            setattr(cls._meta.concrete_model, related.get_accessor_name(), self.related_accessor_class(related))
```

Comment:
If we were to attach the descriptor on concrete model we might have to do adjustments in related field clash checks.